keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Applicationed.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ *
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.spinnaker.keel.api
+
+/**
+ * An object which is associated with an application by the field application
+ */
+interface Applicationed {
+  val application: String
```

Comment:
This seems like a different approach to what we've done with typed metadata before (`name`, `uid`, `serviceAccount`). Curious why the difference.

Reply:
The main difference is this is for the spec, before it gets turned into a resource. This is used to pull the app value into the top level metadata so that we can use it elsewhere. This is used in `resolvableResourceHandler.normalize()`. I didn't want to add another required field to the submittedMetadata when that information is already present in the resource body, just in two different places (some things don't have a moniker, like images).

I can't think of a more elegant way to do this, @cfieber got any suggestions.

Reply:
I like this approach (maybe hung up on the interface name)

Reply:
Yeah, I like it. And we could implement the interface still backed by a metadata field (for JSON consistency, interface field would be ignored).